packages/dotnet-sdk/src/TeamsFx/TeamsFx.cs

```diff
@@ -148,10 +153,15 @@ public async Task<AuthenticationConfiguration> GetAuthenticationConfigurationAsy
         /// </summary>
         /// <param name="credential">Token credential instance.</param>
         /// <param name="scopes">The string of Microsoft Token scopes of access separated by space. Default value is `.default`.</param>
+        /// <param name="logger">Logger of MsGraphAuthProvider class. If the value is null, it will use the logger constructed by DI during TeamsFx class initialization.</param>
         /// <returns>Graph client with specified scopes.</returns>
-        public static GraphServiceClient CreateMicrosoftGraphClient(TokenCredential credential, string scopes = ".default")
+        public GraphServiceClient CreateMicrosoftGraphClient(TokenCredential credential, string scopes = ".default", ILogger<MsGraphAuthProvider> logger = null)
         {
-            var authProvider = new MsGraphAuthProvider(credential, scopes);
+            if (logger == null)
+            {
+                logger = _authLogger;
+            }
+            var  authProvider = new MsGraphAuthProvider(credential, scopes, logger);
```

Comment:
minor: you can use null-coalescing operator ??

Reply:
fixed.